### PR TITLE
Add Bucket Attributes

### PR DESCRIPTION
**Pull request by RachitSharma2001**
Addresses #1025

Allows for the storage, retrieval, and update of bucket attributes. So far, the only bucket attributes that are added are `versioningEnabled` and `defaultEventBasedHolds`. In addition, `versioningEnabled` cannot be updated within the file system, since the file system does not support it. 

Relevant changes to the backend:
1. Added `BucketAttributes` struct -> `internal/backend/bucket.go`
2. Changed `CreateBucket` of filesystem and memory to take in bucket attributes -> `internal/backend/fs.go` and `internal/backend/memory.go`
3. Added `UpdateBucket` method ->` internal/backend/fs.go` and `internal/backend/memory.go`

Relevant changes to fakestorage:
1. Added `updateBucket `endpoint -> `bucket.go` and `response.go`

Relevant changes to GRPC:
1. Added `UpdateBucket` endpoint ->` internal/grpc/server.go`
2. Changed `InsertBucket` to allow the setting of defaultEventBasedHold -> `internal/grpc/server.go`

Relevant tests:
internal/backend/backend_tests.go -> `TestBucketAttrsStoreRetrieveUpdate`, `TestBucketAttrsUpdateVersioning`
fakestorage/bucket_test.go -> `TestServerClientStoreAndRetrieveBucketAttrs`, `TestServerClientUpdateBucketAttrs`
internal/grpc/server_test.go -> `TestBucketInsertGetListUpdateDelete`

### CODE REVIEW

**Comment by fsouza** [comment]
@RachitSharma2001 thanks for working on this and for the detailed PR description. I'm gonna review this later today or tomorrow!

File: fakestorage/bucket.go [hunk 1]
```diff
@@ -24,17 +25,46 @@ var bucketRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9]$`)
...
+	err := s.backend.UpdateBucket(bucketName, attrsToUpdate)
+	if err != nil {
+		panic(err)
+	}
+	return jsonResponse{}
+}
+
+func getBucketAttrsToUpdate(body io.ReadCloser) backend.BucketAttrs {
+	var data map[string]interface{}
+	json.NewDecoder(body).Decode(&data)
```
Comment (fsouza) [inline fakestorage/bucket.go:46]: Check for errors?

**Review by fsouza — COMMENTED** [review]
I think this looks good overall, I gave it a first pass, but plan to give it another one later this week. Thank you very much for contributing and sorry for the slow review.

**Comment by RachitSharma2001** [comment]
@fsouza No worries, and thank you for the feedback!

I have updated the `getBucketAttrsToUpdate()` function according to the changes requested. Let me know if there is anything else that should be changed.

**Review by fsouza — APPROVED** [review]
Thank you very much!

**Pull request merged**